fe/src/main/java/org/apache/doris/catalog/Catalog.java

```diff
@@ -3573,6 +3592,11 @@ private Table createOlapTable(Database db, CreateTableStmt stmt, boolean isResto
                     PropertyAnalyzer.analyzeDataProperty(stmt.getProperties(), DataProperty.DEFAULT_HDD_DATA_PROPERTY);
                     PropertyAnalyzer.analyzeReplicationNum(properties, FeConstants.default_replication_num);
 
+                    Map<String, String> dynamicPartitionProperties = DynamicPartitionUtil.analyzeDynamicPartition(db, olapTable, properties);
```

Comment:
Here you call `analyzeDynamicPartition` before creating the partition.
And inside the `analyzeDynamicPartition`, you will add this table to the 
`DynamicPartitionScheduler`.
But if creating partition failed, no one remove this table out of `DynamicPartitionScheduler`, which causing memory leak.

You should make sure that every is OK, then you can register this table to the `DynamicPartitionScheduler`. Or you need to find a way to clean it out.

Reply:
> Here you call `analyzeDynamicPartition` before creating the partition.
> And inside the `analyzeDynamicPartition`, you will add this table to the
> `DynamicPartitionScheduler`.
> But if creating partition failed, no one remove this table out of `DynamicPartitionScheduler`, which causing memory leak.
> 
> You should make sure that every is OK, then you can register this table to the `DynamicPartitionScheduler`. Or you need to find a way to clean it out.

When DynamicPartitionScheduler begin to execute, dynamicAddPartition function will traverse all tables and remove invalid tables.

Reply:
After you have executed `analyzeDynamicPartition()` and added the table to the `DynamicPartitionScheduler`, but before it was officially added to the database, the `DynamicPartitionScheduler` may starts, and it cannot find this table in database, so it will remove the table.

So better add it after the table is truly created.

Reply:
> After you have executed `analyzeDynamicPartition()` and added the table to the `DynamicPartitionScheduler`, but before it was officially added to the database, the `DynamicPartitionScheduler` may starts, and it cannot find this table in database, so it will remove the table.
> 
> So better add it after the table is truly created.

Done